pkg/trace/agent/agent_test.go

```diff
@@ -761,9 +763,81 @@ func TestFilteredByTags(t *testing.T) {
 			span:    pb.Span{Meta: map[string]string{"something": "else", "key-only": "but-also-value", "bad-key-only": "random"}},
 			drop:    true,
 		},
+		{
```

Comment:
I love the new tests, thanks for doing that. I think it's worth giving each test a name that describes what it's checking. That would make it easier to review and make sure all test cases are covered, and would help with debugging if one of the tests starts failing in the future. If the name gets too long, even just a little comment for each test would be good.

Reply:
I tried to come up with suitable names for the tests. If you have any additional ideas on how to improve the titles, I will do it right away